src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/ECDsa/ECDsaImportExport.cs

```diff
@@ -0,0 +1,347 @@
+                    Assert.True(parameters.Curve.IsNamed);
+                    var q = parameters.Q;
+                    var c = parameters.Curve;
+                    TestNamedImportValidation(ec, ref q, ref c, true);
+
+                    q.X = null; TestNamedImportValidation(ec, ref q, ref c);
+                    q.X = new byte[] { }; TestNamedImportValidation(ec, ref q, ref c);
+                    q.X = new byte[1] { 0x10 }; TestNamedImportValidation(ec, ref q, ref c);
+                    q.X[0]--;  TestNamedImportValidation(ec, ref q, ref c);
+
+                    q.Y = null; TestNamedImportValidation(ec, ref q, ref c);
+                    q.Y = new byte[] { }; TestNamedImportValidation(ec, ref q, ref c);
+                    q.Y = new byte[1] { 0x10 }; TestNamedImportValidation(ec, ref q, ref c);
+                    q.Y[0]--; TestNamedImportValidation(ec, ref q, ref c);
+
+                    parameters.Curve = ECCurve.CreateFromOid(new Oid("Invalid", "Invalid"));
+                    // Ideally this throws PlatformNotSupportedException, but underlying code does not know
+                    //  curve name is not supported
+                    Assert.ThrowsAny<CryptographicException>(() => ec.ImportParameters(parameters));
+                }
+            }
+        }
+
+        [ConditionalFact(nameof(ECExplicitCurvesSupported))]
+        public void TestGeneralExportWithExplicitParameters()
+        {
+            using (ECDsa ecdsa = ECDsaFactory.Create())
+            {
+                ECParameters param = ECDsaTestData.GetNistP256ExplicitTestData();
+                param.Validate();
+                ecdsa.ImportParameters(param);
+                Assert.True(param.Curve.IsExplicit);
+
+                param = ecdsa.ExportParameters(false);
+                param.Validate();
+
+                // We should have explicit values, not named, as this curve has no name.
+                Assert.True(param.Curve.IsExplicit);
+            }
+        }
+
+        private void TestNamedImportValidation(ECDsa ec, ref ECPoint q, ref ECCurve c, bool success = false)
+        {
+            ECParameters p = new ECParameters();
+            p.Q = q;
+            p.Curve = c;
+            if (success)
+            {
+                ec.ImportParameters(p);
+            }
+            else
+            {
+                Assert.ThrowsAny<CryptographicException>(() => ec.ImportParameters(p));
+            }
+            p = ECDsaTestData.GetNistP224KeyTestData();
+            q = p.Q;
+            c = p.Curve;
+        }
+
+        [ConditionalFact(nameof(ECDsa224Available))]
+        public void TestExplicitKey()
+        {
+            using (ECDsa ec = ECDsaFactory.Create())
+            {
+                ECParameters parameters = ECDsaTestData.GetNistP224KeyTestData();
+                ec.ImportParameters(parameters);
+                VerifyNamedCurve(parameters, ec, 224, true);
```

Comment:
Yeah, I totally missed the implication of "Key" there.  My bad; but I like the new name :)